metadata/src/main/java/org/apache/kafka/controller/BrokersToIsrs.java

```diff
@@ -326,4 +326,14 @@ PartitionsOnReplicaIterator partitionsWithBrokerInIsr(int brokerId) {
     boolean hasLeaderships(int brokerId) {
         return iterator(brokerId, true).hasNext();
     }
+
+    int offlinePartitionCount() {
```

Comment:
We cannot count through all the offline partitions this way every time metrics get fetched.  As much as possible, there should be no `O(num_partitions)` operations in metrics collection.